# FASTQ to VCF Pipeline using GATK best practices

This repository is a guide to create a fastq to vcf pipeline using GATK best practices for preprocessing raw high throughput sequencing data from NGS platforms like Illumina to generate a variant call file which can be used for further downstream analysis.

## Laptop configuration

   * Laptop: ROG Strix G15
   * Processor: Intel(R) Core(TM) i5-10300H CPU @ 2.50GHz (4 cores)
   * RAM: 8 GB
   * System type: 64-bit operating system, x64-based processor
   * Storage: 1 TB
   * OS: Windows 11

## WSL2 Installation

* About Windows Subsystem for Linux (WSL): https://learn.microsoft.com/en-us/windows/wsl/about
* Installation steps: https://learn.microsoft.com/en-us/windows/wsl/install

## Bioinformatics Tools

* BWA (Burrows Wheeler Aligner)
   * Usage: This tool is used for aligning sequencing reads (from FASTQ files) to a reference genome (like the human genome in fasta format). It's particularly efficient for mapping short and long DNA reads generated by platforms like Illumina. For more details: https://bio-bwa.sourceforge.net/
   * Installation: Please follow either one of the tutorials for installing BWA
        1. Using Source code: https://youtu.be/oCEqS42N_7E?si=BVm627bmjaSMpzpx
        2. Using Binaries: https://youtu.be/HBu6atfdZK4?si=IOoFpQnnmQay5t9N
   * Input and Output files:
        1. Input: FASTQ files (reads), FASTA (reference)
        2. Output: SAM file

* Samtools
    * Usage:
      In this pipeline samtools is mainly used to convert a SAM (Sequence Alignment Map) file to a sorted BAM (Binary Alignment BAM) file. For more details: https://www.htslib.org/
    * Installation:
        1. Using Source code: https://youtu.be/ULQU8X3abxY?si=MEQSTPjYaNQLRM85
        2. Using Binaries: https://youtu.be/zauS4C_RtZw?si=oDBRdkPpagM9Jkuu
    * Input and Output files:
        1. Input: SAM file
        2. Output: BAM file
  





